src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -254,10 +270,10 @@ class CompanyStep extends React.Component {
                         <View style={[styles.flex1, styles.mr2]}>
                             <DatePicker
                                 label={this.props.translate('companyStep.incorporationDate')}
-                                onChange={value => this.clearErrorAndSetValue('incorporationDate', value)}
+                                onChange={this.clearIncorporationDateErrorsAndSetValue}
                                 value={this.state.incorporationDate}
                                 placeholder={this.props.translate('companyStep.incorporationDatePlaceholder')}
-                                errorText={this.getErrorText('incorporationDate')}
+                                errorText={this.getErrorText('incorporationDate') || this.getErrorText('incorporationDateFuture')}
```

Comment:
Have you tested this? I don't think `this.getErrorText('incorperationDate')` will ever return a falsey value (pretty sure it'll always return the error text string, even if that error isn't happening).

Reply:
Yes I tested and works as expected.  

Basically below line will return error text related to _incorporationDate_ or _incorporationDateFuture_
`errorText={this.getErrorText('incorporationDate') || this.getErrorText('incorporationDateFuture')}` So we do not need true or false value for this.

So if `this.getErrorText('incorporationDate')` return error text then it will show that error, otherwise it will show error text received from `this.getErrorText('incorporationDateFuture')`. 

e.g. Whenever entered date is more than thousand year from current date, it shows error text from this.getErrorText('incorperationDate') e.g. **Please enter a valid date**  (this logic was already there). 

e.g. Another case is entered date is future date (but not more than thousand year from now) so in that case it will show error message from `this.getErrorText('incorporationDateFuture')` e.g. **Incorporation date cannot be in the future**

I also recorded screen for this implementation, here it is: https://github.com/Expensify/App/pull/5939#discussion_r732077137  it is working as expected.

Reply:
Ah, OK. I just looked at getErrorText and saw that it returns an empty string when the error isn't occurring. I think the naming here was confusing for me (I thought it would go look up the localized error text for a given error name).